### ISSUE TITLE
Can't configure Dyson Pure Hot+Cool Link (wrong device type discovered)

### ISSUE THREAD

**Issue by anarcometin** [shenxn/ha-dyson#41]
I have followed the configuration instructions, but I don't seem to be able to configure this integration properly. 

### System details:
* Docker Home Assistant 2021.4.5
* ha-dyson (attempts with versions v0.10.0 and v0.11.0)
* Installation using HACS

**Note: it seems that the wrong device type is getting discovered (360 Eye robot vacuum, instead Pure Hot+Cool link). To be very clear, I don't have any other Dyson devices than this fan.**
 
### Setup manually 
Credentials acquired using [https://github.com/shenxn/libdyson](https://github.com/shenxn/libdyson)
<img width="731" alt="Screen Shot 2021-04-22 at 12 08 58 pm" src="https://user-images.githubusercontent.com/3376956/115651046-c8c44200-a36d-11eb-8abb-9d6663b4ad23.png">
Log message:
`DEBUG (MainThread) [custom_components.dyson_local.config_flow] Failed to connect to device:`

### Setup using device WiFi information logs
Log message:
```
21-04-22 13:08:12 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Successfully parse WiFi information                                                                                                                                                                                                                                                                            
2021-04-22 13:08:12 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Serial: XXX-AU-XXXXXXXX                                                                                                                                                                                                                                                                                        
2021-04-22 13:08:12 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Device Type: N223                                                                                                                                                                                                                                                                                              
2021-04-22 13:08:12 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Device Type Name: 360 Eye robot vacuum                                                                                                                                                                                                                                                                         
2021-04-22 13:08:12 DEBUG (zeroconf-ServiceBrowser__360eye_mqtt._tcp.local.-_dyson_mqtt._tcp.local._326) [custom_components.dyson_local.config_flow] Found device at XXX.XXX.X.XXX                                                                                                                                                                                                                
2021-04-22 13:08:31 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Failed to connect to device:  
```  

Comparing devices numbers from the credential process (**455**) does not seem to match the one in the ha-dyson configuration logs (**N223**).

I have attempted this process selecting either the correct device type (Pure Hot+Cool Link) and the vacuum type as well, always getting `Failed to connect to device`.

Please let me know what am I doing wrong or missing.

**Comment by shenxn** [comment]
What is your device WiFi SSID? You can mask the serial part.

**Comment by anarcometin** [comment]
DYSON-AAN-AA-AAANNNNA-455A

**Comment by shenxn** [comment]
Is this a pretty new device? Since 455A is not in the device type map yet.

**Comment by shenxn** [comment]
Version 0.13.0 should have fixed this. Please try re-add device.

**Comment by shenxn** [comment]
Closing for now. Feel free to reopen if the problem still existed.

**Comment by chris-chensy** [comment]
Hi shenxn,

Thanks for the integration. I'm having the exact some issue with the setup. Below are the logs:

2021-06-19 00:29:14 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Successfully parse WiFi information
2021-06-19 00:29:14 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Serial: SZ1-AU-NEA1727A
2021-06-19 00:29:14 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Device Type: 455A
2021-06-19 00:29:14 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Device Type Name: Pure Hot+Cool Link (New)
2021-06-19 00:29:15 DEBUG (zeroconf-ServiceBrowser__360eye_mqtt._tcp.local.-_dyson_mqtt._tcp.local._266) [custom_components.dyson_local.config_flow] Found device at 192.168.86.200
2021-06-19 00:29:15 DEBUG (Thread-15) [libdyson.dyson_device] Connected with result code 0
2021-06-19 00:29:15 INFO (MainThread) [libdyson.dyson_device] Connected to device SZ1-AU-NEAxxxxx
2021-06-19 00:29:35 DEBUG (Thread-15) [libdyson.dyson_device] Disconnected with result code 0
2021-06-19 00:29:35 DEBUG (MainThread) [custom_components.dyson_local.config_flow] Failed to connect to device:

**Comment by anarcometin** [comment]
Apologies for not reporting back before. I have the same connection issue as @chris-chensy is reporting.
Attempting setup via WiFi information or manual configuration, both end up with a `Failed to connect to device` message.

I can confirm that the WiFi information process is parsing the WiFi SSID to the new Device Type, but it fails connection.

**Comment by shenxn** [comment]
@anarcometin @chris-chensy  Can you try set up through manual configuration and select the old 455 model to see if it works? Thanks.

**Comment by shenxn** [comment]
I've released version 0.15.0. Hopefully it fixes your problem.

**Comment by ariell** [comment]
Thanks for the effort and the integration! I also have not been able to get it to work.  I've been trying to configure for the last few days, but not having any luck. I've updated to 0.15.0, but still no luck.

My SSID is: `DYSON-SZ2-AU-XXXXXXXX-455A`
My country code is: `AU`

Device shows up in dyson-cloud (sometimes, but keeps disappearing every now and then). Sometimes `libdyson:get_devices.py` works and sometimes fails. I've tried manual config selecting all pure hot+cool options, and keep getting: 
```
Failed to find the device using discovery
```
or if I fill in the host:
```
Failed to connect to device
```
